cognite_toolkit/cdf_tk/run.py

```diff
@@ -146,3 +158,222 @@ def run_transformation(ToolGlobals: CDFToolConfig, external_ids: str | list[str]
             print(f"[bold red]ERROR:[/] Could not run transformation {transformation.external_id}.")
             print(e)
     return True
+
+
+def run_local_function(
+    ToolGlobals: CDFToolConfig,
+    source_path: Path,
+    external_id: str,
+    payload: str,
+    build_env: str,
+    schedule: str | None = None,
+    rebuild_env: bool = False,
+    verbose: bool = False,
+    no_cleanup: bool = False,
+) -> bool:
+    system_config = SystemConfig.load_from_directory(source_path / COGNITE_MODULES, build_env)
+    config = BuildConfigYAML.load_from_directory(source_path, build_env)
+    print(f"[bold]Building for environment {build_env} using {source_path!s} as sources...[/bold]")
+    config.set_environment_variables()
+    build_dir = tempfile.mkdtemp(prefix="build.", suffix=".tmp", dir=Path.cwd())
+    found = False
+    for root_module, root_values in config.modules.items():
+        if not isinstance(root_values, dict):
+            continue
+        for group_module, group_values in root_values.items():
+            if not isinstance(group_values, dict):
+                continue
+            for module_name in group_values:
+                if Path(source_path / f"{root_module}/{group_module}/{module_name}/functions/{external_id}").is_dir():
+                    config.environment.selected_modules_and_packages = [f"{module_name}"]
+                    found = True
+                    break
+    if not found:
+        print(f"  [bold red]ERROR:[/] Could not find function with external id {external_id}, exiting.")
+        return False
+    build_config(
+        build_dir=Path(build_dir),
+        source_dir=source_path,
+        config=config,
+        system_config=system_config,
+        clean=True,
+        verbose=False,
+    )
+    environment_dir = Path(source_path / f"./.venv.{external_id}")
```

Comment:
Since we already have the concept of `environment` in the toolkit. 
```suggestion
    virtual_env_dir = Path(source_path / f"./.venv.{external_id}")
```